tests/contrib/molten/test_molten.py

```diff
@@ -0,0 +1,171 @@
+from unittest import TestCase
+
+import molten
+from molten.testing import TestClient
+
+from ddtrace import Pin
+from ddtrace.propagation.http import HTTP_HEADER_TRACE_ID, HTTP_HEADER_PARENT_ID
+from ddtrace.contrib.molten import patch, unpatch, MOLTEN_VERSION
+
+from ...test_tracer import get_dummy_tracer
+from ...util import override_config
+
+# NOTE: Type annotations required by molten otherwise parameters cannot be coerced
+def hello(name: str, age: int) -> str:
+    return f'Hello {age} year old named {name}!'
+
+
+def molten_client(headers=None):
+    app = molten.App(routes=[molten.Route('/hello/{name}/{age}', hello)])
+    client = TestClient(app)
+    uri = app.reverse_uri('hello', name='Jim', age=24)
+    if headers:
+        return client.request('GET', uri, headers=headers)
+    return client.get(uri)
+
+
+class TestMolten(TestCase):
+    """"Ensures Molten is properly instrumented."""
+
+    TEST_SERVICE = 'molten-patch'
+
+    def setUp(self):
+        patch()
+        self.tracer = get_dummy_tracer()
+        Pin.override(molten, tracer=self.tracer)
+
+    def tearDown(self):
+        unpatch()
+        self.tracer.writer.pop()
+        delattr(self, 'tracer')
+
+    def test_route_success(self):
+        """ Tests request was a success with the expected span tags """
+        response = molten_client()
+        spans = self.tracer.writer.pop()
+        self.assertEqual(response.status_code, 200)
+        # TestResponse from TestClient is wrapper around Response so we must
+        # access data property
+        self.assertEqual(response.data, '"Hello 24 year old named Jim!"')
+        span = spans[0]
```

Comment:
we should probably also assert the span created for the view function.

Reply:
Again, I think `test_resources` addresses this case.

Reply:
we should probably assert on the span name and the span resource, and for the exception cases we want to assert that it is marked as an error with the exception information.

testing here would just be `span.name`, `span.resource`, and `span.error == 0`